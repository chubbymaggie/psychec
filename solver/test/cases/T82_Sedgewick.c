//#include <stdlib.h>
//#include "GRAPH.h"
//struct graph { int V; int E; int **adj; };
Graph GRAPHinit(int V)
  { Graph G = malloc(sizeof *G);
    G->V = V; G->E = 0;
    G->adj = MATRIXint(V, V, 0);
    return G;
  }
void GRAPHinsertE(Graph G, Edge e)
  { int v = e.v, w = e.w;
    if (G->adj[v][w] == 0) G->E++;
    G->adj[v][w] = 1; 
    G->adj[w][v] = 1;
  }
void GRAPHremoveE(Graph G, Edge e)
  { int v = e.v, w = e.w;
    if (G->adj[v][w] == 1) G->E--;
    G->adj[v][w] = 0; 
    G->adj[w][v] = 0;
  }
int GRAPHedges(Edge a[], Graph G)
  { int v, w, E = 0;
    for (v = 0; v < G->V; v++)
      for (w = v+1; w < G->V; w++)
        if (G->adj[v][w] == 1) 
          a[E++] = EDGE(v, w); 
    return E;
  }
