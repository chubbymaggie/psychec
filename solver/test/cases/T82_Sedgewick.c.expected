/* Forward declarations */
typedef  struct TYPE_13__   TYPE_2__ ;
typedef  struct TYPE_12__   TYPE_1__ ;

/* Type definitions */
struct TYPE_13__ {int v; int w; } ;
struct TYPE_12__ {int V; int** adj; scalar_t__ E; } ;
typedef  TYPE_1__* Graph ;
typedef  TYPE_2__ Edge ;

/* Variables and functions */
 TYPE_2__ EDGE (int,int) ;
 int** MATRIXint (int,int,int /*<<< orphan*/ ) ;
 TYPE_1__* malloc (int) ;
